airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/destination.py

```diff
@@ -3,53 +3,98 @@
 #
 
 
-import os
+import tempfile
+from logging import Logger
+from pathlib import Path
 from typing import Any, Iterable, Mapping, Optional
 
-from airbyte_cdk import AirbyteLogger
+from airbyte.secrets import SecretString
+from airbyte.strategies import WriteStrategy
 from airbyte_cdk.destinations import Destination
-from airbyte_cdk.destinations.vector_db_based.embedder import Embedder, create_from_config
-from airbyte_cdk.destinations.vector_db_based.indexer import Indexer
-from airbyte_cdk.destinations.vector_db_based.writer import Writer
-from airbyte_cdk.models import AirbyteConnectionStatus, AirbyteMessage, ConfiguredAirbyteCatalog, ConnectorSpecification, Status
-from airbyte_cdk.models.airbyte_protocol import DestinationSyncMode
+from airbyte_cdk.destinations.vector_db_based.embedder import (
+    Embedder,
+    FakeEmbedder,
+    FakeEmbeddingConfigModel,
+)
+from airbyte_cdk.models import (
+    AirbyteConnectionStatus,
+    AirbyteMessage,
+    ConfiguredAirbyteCatalog,
+    ConnectorSpecification,
+    DestinationSyncMode,
+    Status,
+)
+
+from destination_snowflake_cortex import cortex_processor
+from destination_snowflake_cortex.common.catalog.catalog_providers import CatalogProvider
 from destination_snowflake_cortex.config import ConfigModel
-from destination_snowflake_cortex.indexer import SnowflakeCortexIndexer
 
 BATCH_SIZE = 150
 
 
 class DestinationSnowflakeCortex(Destination):
-    indexer: Indexer
     embedder: Embedder
+    sql_processor: cortex_processor.SnowflakeCortexSqlProcessor
 
-    def _init_indexer(self, config: ConfigModel, configured_catalog: Optional[ConfiguredAirbyteCatalog] = None):
-        self.embedder = create_from_config(config.embedding, config.processing)
-        self.indexer = SnowflakeCortexIndexer(config.indexing, self.embedder.embedding_dimensions, configured_catalog)
+    def _init_sql_processor(
+        self, config: ConfigModel, configured_catalog: Optional[ConfiguredAirbyteCatalog] = None
+    ):
+        self.sql_processor = cortex_processor.SnowflakeCortexSqlProcessor(
```

Comment:
Indexer is replaced by SqlProcessor, and Embedding functions are moved into the SqlProcessor so that they can be performed more efficiently: in bulk, rather than on a per-record/per-document basis. This also means that we never have to create multiple record objects for the same source records, and we don't have to create an adapted catalog. The source catalog is just the source catalog, and the storage of the record into different columns is handled internally by the SqlProcessor.